### PR TITLE
feat: status master and status workflow added

### PR DESCRIPTION
**Pull request by vishdha**

### CODE REVIEW

File: frappe/core/doctype/status_workflow/status_workflow.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  },
+  {
+   "fieldname": "section_break_5",
+   "fieldtype": "Section Break"
+  },
+  {
+   "description": "Optional: The status will be set if this expression is true",
+   "fieldname": "condition",
+   "fieldtype": "Code",
+   "ignore_xss_filter": 1,
```
Comment (hrwX) [inline frappe/core/doctype/status_workflow/status_workflow.json]:
```suggestion
```

File: frappe/core/doctype/status_workflow/status_workflow.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from __future__ import unicode_literals
+import frappe
+import json, os
+from frappe.utils import nowdate
+from frappe import _
+from frappe.model.document import Document
+
+class StatusWorkflow(Document):
+	def autoname(self):
+		self.name = self.document_type+ "-" + self.status
```
Comment (hrwX) [inline frappe/core/doctype/status_workflow/status_workflow.py:14]: - Put this in auto_name in doctype itself

File: frappe/core/doctype/status_workflow/status_workflow.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		if self.is_standard and not frappe.conf.developer_mode:
+			frappe.throw(_('Cannot edit Standard Status Workflow. To edit, please disable this and duplicate it'))
+	
+	# def validate_condition(self):
+	# 	temp_doc = frappe.new_doc(self.document_type)
+	# 	if self.condition:
+	# 		try:
+	# 			frappe.safe_eval(self.condition, None, get_context(temp_doc.as_dict()))
+	# 		except Exception:
+	# 			frappe.throw(_("The Condition '{0}' is invalid").format(self.condition))
```
Comment (hrwX) [inline frappe/core/doctype/status_workflow/status_workflow.py]:
```suggestion
```

File: frappe/core/doctype/status/status.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+# import frappe
+from frappe.model.document import Document
+
+class Status(Document):
```
Comment (hrwX) [inline frappe/core/doctype/status/status.py:9]: - Are we not moving the status fields of existing doctypes to Status doctype ?

**Review by hrwX — CHANGES_REQUESTED** [review]

File: frappe/core/doctype/status/status.py (hunk 4 above)
Comment (vishdha) [reply frappe/core/doctype/status/status.py:9]: we replacing select field to link

File: frappe/core/doctype/status_workflow/status_workflow.py (hunk 2 above)
Comment (vishdha) [reply frappe/core/doctype/status_workflow/status_workflow.py:14]: let be it should be coded here. the front end will give the control to the user for change in formating. there is no issue right now but let be coded here

Comment (hrwX) [reply frappe/core/doctype/status_workflow/status_workflow.py:14]: This does not make sense to hardcode something in code that can be achieved with configuration

File: frappe/core/doctype/status/status.py (hunk 4 above)
Comment (hrwX) [reply frappe/core/doctype/status/status.py:9]: I cannot see the json for doctypes in the framework for which we changed select field to link

**Review by hrwX — CHANGES_REQUESTED** [review]

**Comment by sahil28297** [comment]
@vishdha what's up with this PR?

File: frappe/core/doctype/status/status.py (hunk 4 above)
Comment (vishdha) [reply frappe/core/doctype/status/status.py:9]: its in erpnext

**Pull request closed without merging**